### ISSUE TITLE
Support for ZTE Zmax Pro....Bayer file uploaded

### ISSUE THREAD

**Issue by aggere1** [troop/FreeDcam#126]
Hello.  I've read where if I cannot see the option to convert raw bayer file to DNG that you request that the raw bayer file be provided so you can add support.  I have uploaded an example bayer file here:

http://s000.tinyupload.com/index.php?file_id=00117460921766582962

Thank you.

**Comment by defcomk** [comment]
in the menu  click on save camera parameters and upload the text file that gets saved /DCIM/FreeDcam also using ES File Explorer or Root Explorer navigate to /System/ Lib/HW/ and look for a file that starts with camera.msm89 and upload that also wanna check if the zMax uses similar Extensions to the nubia devices

**Comment by aggere1** [comment]
[Z981_CameraParameters.txt](https://github.com/troop/FreeDcam/files/669972/Z981_CameraParameters.txt)

SO file is here:
http://s000.tinyupload.com/index.php?file_id=00001160122446523481

Thanks for your help!

**Comment by aggere1** [comment]
Ha!  did not realize I could close an issue....sorry.  lol.

**Comment by aggere1** [comment]
Did I post the correct information?

**Comment by defcomk** [comment]
yeah you did will post a test apk in a couple of hours

**Comment by aggere1** [comment]
Oh OK.  Thanks!

On Dec 23, 2016 12:53 PM, "defcomk" <notifications@github.com> wrote:

yeah you did will post a test apk in a couple of hours

—
You are receiving this because you modified the open/close state.

Reply to this email directly, view it on GitHub
<https://github.com/troop/FreeDcam/issues/126#issuecomment-269024632>, or mute
the thread

**Comment by defcomk** [comment]
[app-debug.zip](https://github.com/troop/FreeDcam/files/671578/app-debug.zip)

**Comment by aggere1** [comment]
application crashes.  first time I accessed the apk file it asked me if I wanted to update installation and I said yes.  after granting permissions it crashed.  I also tried uninstalling the application, rebooting and then installing the apk.  same result....

**Comment by defcomk** [comment]
logcat the fc

**Comment by aggere1** [comment]
I'm sorry, what?

**Comment by aggere1** [comment]
disregard prior post...I googled adb, installed SDK tools, etc.

I'm doing something wrong though....after struggling to get my device listed, etc. I'm not successfully producing a log file with:

adb logcat -d -f /sdcard/logcat.log *:D

I've done this both before and after the application crash....I'm not creating a log file logcat.log on my sdcard....there is a file in the root directory of sdcard called USERRETURN file....the contents of this file are:

com.estrongs.android.popfi	1482445083669
com.estrongs.android.popvc	552

I suspect I'm not terminating the adb session correctly....am I just supposed to Ctrl-C after entering the adb statement above?

Thanks.

**Comment by troop** [comment]
`adb logcat -d > logcat.txt` save log to the folder where adb is

**Comment by aggere1** [comment]
Well, I guess I'm making progress.  The app did not crash, and created a bayer file, but there is no option to convert to DNG.

Is the test app version that you gave 3.3.147 (103)?  I have had different versions on my phone and want to be sure that I'm logging the correct version (!).

[logcat.txt](https://github.com/troop/FreeDcam/files/671810/logcat.txt)

Bayer file: http://s000.tinyupload.com/index.php?file_id=00512791018453723660

**Comment by aggere1** [comment]
Play store pushed 3.3.147 to my phone....I uninstalled, cleared phone cache for all apps, rebooted and installed the test file you provided above.

Crash is back.  I also failed to mention that the viewer program is glitching....attaching two screenshots of the viewer glitches and the logcat file which includes both the freedcam crash and the attempted opening of the viewer app....

http://s000.tinyupload.com/index.php?file_id=46086322341716817050
http://s000.tinyupload.com/index.php?file_id=09635231794642379387
[logcat.txt](https://github.com/troop/FreeDcam/files/671887/logcat.txt)

**Comment by troop** [comment]
the apk you have contains defcomk latest changes. they did not get merged into 3.3.147

crash is that:

`12-24 00:37:07.889  4792  4792 E AndroidRuntime: java.lang.NullPointerException: Attempt to invoke interface method 'java.lang.String freed.cam.apis.basecamera.parameters.modes.ModeParameterInterface.GetValue()' on a null object reference
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at freed.cam.ui.themesample.cameraui.CameraUiFragment.onViewCreated(CameraUiFragment.java:303)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1127)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1290)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:801)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.app.FragmentManagerImpl.execSingleAction(FragmentManager.java:1638)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.app.BackStackRecord.commitNowAllowingStateLoss(BackStackRecord.java:679)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.app.FragmentStatePagerAdapter.finishUpdate(FragmentStatePagerAdapter.java:166)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.view.ViewPager.populate(ViewPager.java:1240)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.view.ViewPager.populate(ViewPager.java:1088)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v4.view.ViewPager.onMeasure(ViewPager.java:1614)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.View.measure(View.java:18805)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.RelativeLayout.measureChildHorizontal(RelativeLayout.java:716)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:462)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.View.measure(View.java:18805)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6002)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.support.v7.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:135)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.View.measure(View.java:18805)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6002)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1465)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:748)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:630)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.View.measure(View.java:18805)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6002)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.View.measure(View.java:18805)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6002)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1465)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.LinearLayout.measureVertical(LinearLayout.java:748)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.LinearLayout.onMeasure(LinearLayout.java:630)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.View.measure(View.java:18805)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:6002)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.widget.FrameLayout.onMeasure(FrameLayout.java:194)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at com.android.internal.policy.PhoneWindow$DecorView.onMeasure(PhoneWindow.java:2661)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.View.measure(View.java:18805)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:2105)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1221)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1457)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1112)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6050)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.Choreographer.doCallbacks(Choreographer.java:670)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.Choreographer.doFrame(Choreographer.java:606)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.os.Handler.handleCallback(Handler.java:743)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.os.Handler.dispatchMessage(Handler.java:95)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.os.Looper.loop(Looper.java:171)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at android.app.ActivityThread.main(ActivityThread.java:5417)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at java.lang.reflect.Method.invoke(Native Method)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
12-24 00:37:07.889  4792  4792 E AndroidRuntime: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)`

**Comment by aggere1** [comment]
So what does that mean.... As it relates to ultimately being able to
successfully create DNG files from Bayer files?

**Comment by troop** [comment]
[freedcam3.3.147beta1-debug.apk.zip](https://github.com/troop/FreeDcam/files/673177/freedcam3.3.147beta1-debug.apk.zip)
that is based on 3.3.147 and the changes defcomk made for your devices. you should see now dng in picformats

**Comment by aggere1** [comment]
DNG functions perfectly now.

Other parameters in settings say Test... I'm assuming by design and would
be removed when this makes into next (?) release... Yes?

Freedcam viewer still showing glitches (see screens above).

In Snapseed photo app, the DNG file file info says f/1 versus correct f/2.2.

Thanks!!!

**Comment by aggere1** [comment]
Disregard comments re test parameters above. 

Bracketing produces images with equal exposure... Known issue?

**Comment by defcomk** [comment]
Adjust it the menu under bracketing you should see 0 0 0

**Comment by aggere1** [comment]
I activated bracketing and set bracketing in menu as -2, -1, 0, 1, 2 on the first five values.

In RawTherapee, the Exposure of the first 5 files was 1/30....but the ISO values were 465, 508, 589, 683, and 821.  I don't know if these values are correct or just an artifact of how the metadata is being read (?)....because there are also strange values in the "quick info" image box in RawTherapee....

![quickimageinfo](https://cloud.githubusercontent.com/assets/24722728/21487888/fc2b7a64-cba4-11e6-90db-7648a4d185f0.JPG)


By the way, I am unable to change the storage location to External in the menu....that's where I was seeing "Test" for several of the menu variables....whenever I set storage to root of SD card in android settings (sent there by freedcam internal/external menu choice) when I return to the app there are many fields that say test and I'm basically stuck in the menu....when I exit and relaunch the app....the "internal" field value has returned....

**Comment by troop** [comment]
ae bracket is up to ae as it says^^. it set the exposure compensation to the given values.
so when -12 is for you iso465 and shutter 1/30  -10 iso508 1/30 i would say that your device use fixed framerate. i recommend to try that stuff with jpegs for checking exif. dng can have incorrect/none exif informations due the camera1 api.

and storage detection is a hacky implementation. android did not offical support external storages till L.
so when your external sd is not mounted in the path i expect it fails. the current workaround seemed till now to work

**Comment by aggere1** [comment]
Thanks for all your help.

The viewer was working when I first installed freedcam but the problem
still occurs even if I Uninstall purge cache install old working version
etc... So I suspect there is a file or files that don't Uninstall properly
but this is not a big problem so I'm giving up on the viewer.

Again thanks to you and Defcom.

**Comment by aggere1** [comment]
Although I'm able to create DNG files, the DNGs that are produced are not compliant with the saved parameters....specifically the DNGs are all "high ISO" files regardless of whether I choose Auto ISO, ISO100, IS0400, ISOmanual.  

To test this, I took identical JPG and DNG shots at ISO400 and auto exposure, pausing only to change from JPG to DNG, with the following (unexpected) results:

JPG output was 1080 x 1920 at ISO400 (which agrees with saved parameters).
DNG output was 3136 x 4208 at ISO1345 (resolution and ISO should have matched the JPG).

Exposure times (again, used auto exposure) were 1/7 and 1/11 sec.  The test version of the app reports the aperature as f/2.2 in JPG exif data but f/0.0 in DNG exif data (the spec for the camera IS f/2.2).

In RawTherapee, here is the quick info about the DNG file....

![capture](https://cloud.githubusercontent.com/assets/24722728/21531790/bbec8504-cd18-11e6-8c4b-1902c3ad0f35.JPG)


Here are the respective JPG and DNG files.....
JPG:
![img_2016_12_28_15_44_01](https://cloud.githubusercontent.com/assets/24722728/21531742/71cee7c8-cd18-11e6-9e58-80665439e90e.jpg)
DNG:
http://s000.tinyupload.com/index.php?file_id=09718880412116024276

**Comment by aggere1** [comment]
not understanding why freedcam is disregarding the parameters associated with fixed ISO....here is an excerpt from the parameters file....

cur-exposure-time=0.100003

cur-focus-diopter=7.51174
cur-focus-scale=40
**cur-iso=1745**

instant-aec=0
instant-capture=0
internal-restart=true
iso-values=auto,ISO_HJR,ISO100,ISO200,ISO400,ISO800,ISO1600,ISO3200
**iso=ISO400**

picture-format-values=jpeg,bayer-mipi-10gbrg,bayer-mipi-10grbg,bayer-mipi-10rggb,bayer-mipi-10bggr,yuv422sp,bayer-mipi-12gbrg,bayer-mipi-12grbg,bayer-mipi-12rggb,bayer-mipi-12bggr
picture-format=bayer-mipi-10bggr
picture-size-values=4160x3120,4096x2304,3200x2400,3264x1836,2592x1944,2560x1920,2560x1440,2048x1536,1920x1080,1280x720,720x480,640x480,320x240,176x144
picture-size=**4160x3120**

qc-camera-features=8391935
qc-max-num-requested-faces=5
raw-size=**4208x3120**

**Comment by troop** [comment]
that cur-iso and cur-exposuretime shows the true values only after a picture is taken and the preview is not started. at least on the lg g4. (g4 increase iso x2 and lower shutter /2, so with iso 400 you see iso800 and when shutter is1/30 it gets to 1/60)
when preview runs it shows the values applied to the preview. preview trys always to have constant framerate to have a fast autofocus as possible.

but that stuff seems to work different on each phone. depending how the oems stayed on qcom default.

raw size is always the full sensor size. and its normal that the rawsize its bigger then the jpegsize.
due postprocessing interpolating/demosaic it can get cropped.

exiftools tells that about your dng:(cut off matrix stuff etc)

> Exposure Time                   : 1/11
F Number                        : 0
ISO                             : 1345
Aperture Value                  : 1.0
Flash                           : No Flash
Focal Length                    : 3.5 mm
Image Size                      : 4208x3136
Shutter Speed                   : 1/11

**Comment by aggere1** [comment]
So why are the same settings (other than jpg¬>dng) creating an ISO400 jpg
and 1345 dng (when I have set iso at 400)?

On Dec 29, 2016 12:29 AM, "Ingo" <notifications@github.com> wrote:

that cur-iso and cur-exposuretime shows the true values only after a
picture is taken and the preview is not started. at least on the lg g4. (g4
increase iso x2 and lower shutter /2, so with iso 400 you see iso800 and
when shutter is1/30 it gets to 1/60)
when preview runs it shows the values applied to the preview. preview trys
always to have constant framerate to have a fast autofocus as possible.

but that stuff seems to work different on each phone. depending how the
oems stayed on qcom default.

raw size is always the full sensor size. and its normal that the rawsize
its bigger then the jpegsize.
due postprocessing interpolating/demosaic it can get cropped.

exiftools tells that about your dng:(cut off matrix stuff etc)

Exposure Time : 1/11
F Number : 0
ISO : 1345
Aperture Value : 1.0
Flash : No Flash
Focal Length : 3.5 mm
Image Size : 4208x3136
Shutter Speed : 1/11

—
You are receiving this because you modified the open/close state.
Reply to this email directly, view it on GitHub
<https://github.com/troop/FreeDcam/issues/126#issuecomment-269583801>, or mute
the thread

**Comment by aggere1** [comment]
Sorry to stack posts, but I just noticed something....

I reported current ISO 1745 and ISO 400 in the params, and you mentioned that exiftools says ISO 1345.

Is it a coincidence that 1745 - 400 - 1345?

**Comment by aggere1** [comment]
**I figured out what is happening**....

when I switch from jpg to dng, the manual ISO exposure time is disregarded...for reasons I don't understand.

Here is the "experiment" I performed....
set exposure compensation to -2.  auto exposure lock false, auto exposure frame average, exposure time 6 seconds (setting the exposure time should override any of the auto exposure settings).  
Set white balance to manual in the left icon and 4000 in the manual settings.  set brightness to 3 in the manual settings.  Focus and ISO left on auto.

Took a jpg.  exif data reported exposure time 6 seconds (matching manual setting).
Increased exposure time to 12 seconds.  Took a jpg.  exif data exposure time 12 seconds.
exif data for both of these jpgs was ISO100....as expected with the long exposure time....the 12 second shot was almost completely blown out.

Exif data for the 6 second exposure time jpg shot:
[IMG_2016_12_29_16_59_25.txt](https://github.com/troop/FreeDcam/files/678099/IMG_2016_12_29_16_59_25.txt)

Changed the exposure time to 10 seconds.  Took a DNG.
exif data reported exposure time 1/30 seconds.  ISO 357.
Increased exposure time to 30 seconds.  Took a 2nd DNG.
exif data reported exposure time 1/30 seconds.  ISO 357.

Exif data for this dng shot:
[temp.txt](https://github.com/troop/FreeDcam/files/678094/temp.txt)

Switched back to jpg and manually set ISO to 100.  exif data matched 30 seconds and 100 ISO (picture completely white / blown out at this exposure).

Switched back to dng and....you guessed it....exposure time 1/30 seconds, ISO 347.

I should mention that I can clearly see the shutter open for a long time on the jpg exposure times > 1 second, and when I switch to DNG, the shutter simply opens and closes quickly.

So I guess the only question is is it a hardware or a software issue that is causing the failure of the camera to use the correct exposure time when in DNG mode?

**Comment by aggere1** [comment]
The hardware is recognizing manual exposure time when in DNG format mode....but works fine in JPG mode.

**Comment by aggere1** [comment]
Snooping around the stackoverflow website, I'm starting to see how freedcam works...you can query a particular device and determine what it supports (manual exposure, manual ISO, etc) and then you can integrate those features as Camera1 API "extensions" (don't know if I'm using correct terminology here, but that's the jist of it anyway).

So I'm curious if freedcam code specifically turns off automatics like auto-exposure or if the user has to do certain things (like setting auto white balance to manual for example) in order to be able to affect manual changes to ISO, exposure time, etc.

And by extension I'm trying to figure out if there is something specific to DNG / bayer format versus JPG format in how the API....or extensions to the API....handle manual exposure time management.

I also wanted to mention that I saw that manual exposure for my device has FOUR values....off, manual, time priority, iso priority.  Makes me wonder if (a) this is normal for a "Camera1" phone, and (b) if there is something that needs to be added to freedcam code to differentiate manual exposure time from time priority or iso priority.

Forgive me if my idle comments are not adding any value....hopefully they either give you some insight or you can validate that there is something related to these concepts going on....if only for the benefit of my own understanding...

**Comment by troop** [comment]
thats no big deal. its good when users are motivated.
camera1 use a list of strings for the parameters.
overview from different devices is [here](https://github.com/troop/FreeDcam/tree/master/Camera1Parameters)

normaly camera1 should handel AE/q3a for raw and jpeg stream the same way.
your device is not the first wich not acts as expected. redmi note3 snapdragon acts same.
for me it seems that this is a problem from the newer qcom libs. some how it bypass the manualmodes/q3a
q3a = iso/shutter/wb

this is a problem from the camera1 native hal and can only get fixed by the oem or someone trys to mod the lib. the main problem is that raw is not offical supported by android for camera1. so oems do not realy care about and such things happens

> I also wanted to mention that I saw that manual exposure for my device has FOUR values....off, manual, time priority, iso priority

that parameter tells us that you can set manual iso with ae controling the shutter speed(iso priority).
or you can set shuttertime with ae contrling the iso to get a well exposured image.(time priority).
manual gives you full control over shutter and iso.
freedcam handel is that way. aslong you keep iso at auto/0 and only move the shutter slider the time priority is used. same with shutter at auto/0 and iso moved. moving both sliders full manual is active.
both at auto AE is used.

**Comment by aggere1** [comment]
Your explanation was very helpful; thank you.

Do you imagine that I would have to convince ZTE or Qualcomm or both to
correct the qcom libs?  Not sure I understand if we are talking ZTE or
Qualcomm engineers here....
or possibly even third party?

**Comment by aggere1** [comment]
So if the OEMs don't care maybe I'll try myself, lol.

The latest apk you gave me was called 3.3.147beta1-debug.

If I clone 3.3.147beta1 branch to a project in android studio on my desktop, will that include defcom's changes?

**Comment by aggere1** [comment]
**Can the changes reflected in freedcam3.3.147beta1-debug.apk.zip be incorporated into the master?**

You've given me DNG support, just without the manual iso / exposure support I had hoped for (due to ZTE software limitations).

**Assuming the changes can be worked into the master branch, this issue should be closed.**

By the way, if you did not know, ZTE has released open source linux kernel code HERE:

http://opensource.ztedevice.com/

The iso / exposure limitations seem to be visible in msm-camera.h in the include/linux/uapi/media folder (if I'm reading everything correctly)....

**Comment by troop** [comment]
> Can the changes reflected in freedcam3.3.147beta1-debug.apk.zip be incorporated into the master?

the changes are on master
see e3702356e7d9b8aa26327038cc7fc33dd44ee06f
everything after that commit should not get used. im currently reworking appsettings, not all settings get reapplied.

> By the way, if you did not know, ZTE has released open source linux kernel code HERE:

freedcam use the parameters from the camera. when that changes work, a longer max-exposuretime should get returned and is autmatical used. from what i see defcomk did not add the shutterreset logic for your device. that could cause it to not work with dng.

**Comment by aggere1** [comment]
I probably worded my question poorly....sorry....what I meant to ask is when would I expect the successful ability to capture a DNG photo on my ZTE to be added to the deployed google play app?

Also, will you or defcomk add the shutterreset logic?

Many thanks!

**Comment by aggere1** [comment]
Hello.  Sorry to be rude and "bump my own post" but I wanted to confirm that you saw my last two questions?  Thanks!

**Comment by troop** [comment]
yes i read it^^ cant say much when playstore gets updated, its done when its done. im reworking some parts and how stuff get detected. that will take some time depending how much free time i have and how motivated i am.

**Comment by aggere1** [comment]
Understood.  Thanks for the reply.  I plan on repping the app in the ZTE
community when its updated but you are the ones doing all the magic!!

**Comment by aggere1** [comment]
So is the shutter reset logic you mentioned a copy paste exercise from another device or...?

Thanks!

**Comment by troop** [comment]
https://github.com/troop/FreeDcam/blob/master/app/src/main/java/freed/cam/apis/camera1/modules/PictureModule.java#L120

look for these lines and add your device. maybe it will work

**Comment by aggere1** [comment]
I'd love to, but I'm apparently not even capable of successfully setting up android studio to use Ndk 12b in the build of commit  e370235.  I can't even reproduce the apk you've posted previously.

**Comment by troop** [comment]
ndk12b works for me. on ndk13 they switched full to c11 wich dont yet support openmp.
for 64bit:
https://dl.google.com/android/repository/android-ndk-r12b-windows-x86_64.zip
for 32bit
https://dl.google.com/android/repository/android-ndk-r12b-windows-x86.zip

**Comment by aggere1** [comment]
Yes i was able to successfully build and install the debug version that you
and Defcom created.  I will try to hack the code this weekend   thanks.

**Comment by aggere1** [comment]
Well troop I made the suggested changes and successfully built a new apk and tried it...no changes in application behavior....dng format ignores manual ISO and shutter time.

I thought I might try making ZTE_ZMAX_PRO.java look as "comprehensive" (?) as ZTE_ADV.java (the latter imports many more packages and overrides many more methods) but as I looked around in the project structure there is custom logic for ZTE_ADV in dozens of java files....and I don't even know if the capabilities of the two devices are similar so....?

Any other suggestions for me to try?

Thanks!

**Comment by aggere1** [comment]
I contacted Qualcomm and asked for their assistance correcting errors made by ZTE and Xiaomi in HAL library files.

Worst case scenario is they laugh at me.  :)

**Comment by troop** [comment]
I dont think you will get an answer. Camera1 does not offical support raw capture. The problem is already in the qcom source and disable manuals when its a raw stream.

**Comment by aggere1** [comment]
Exactly   the problem is already in the Qualcomm source so they should fix
the source problem.  Anyway it's not like i have any other options.

**Comment by troop** [comment]
you can try to set that to your build.prop

**persist.camera.manual.capture=3** or 4 for multiply raws (dont ask what they mean with that)
for 0,1,2 raw manual gets disabled. but for 2 it should be enabled too wich is not the case

**Comment by aggere1** [comment]
Thanks.

But i think i need root priviledges or boot unlocker to hack build.prop
dont I?  Tmobile/ZTE appear unwilling to allow root access (huge thread on
XDA about it).

**Comment by bebolint98** [comment]
@troop I added this line to the build.prop at my Redmi Note 3 Pro, but still no affect to dng. Choose 0,6 sec shutter but It looks the same as with faster shutter speed. I tried also the persist.camera.manual.capture=1 but the is still there a manual control for raw images.

**Comment by aggere1** [comment]
bebolint98, google says camera2 can be enabled in the redmi note 3 with "'persist.camera.HAL3.enabled=1" then using camera fv-5 etc.  only posting in case you were not aware....

**Comment by bebolint98** [comment]
@aggere1 Yes I already added this to build.prop. Freedcam in camera2 mode it apply the setting to dng. I take a photo in camera1 iso150 0.6sec and in camera2 iso150 0.6 sec and this looks brighter than the cam1 one. Tried Az Camera and it also apply the setting for dng.

**Comment by aggere1** [comment]
Have you tried FV5 and/or Better Camera Unlocked?

**Comment by defcomk** [comment]
persist.camera.rdi.mode=1
persist.camera.offlineraw=1
persist.camera.manual.capture=3

Seems like QC bug xiaomi mi5s has similar issue try the above build.prop params

**Comment by bebolint98** [comment]
@aggere1 At camera fv5 I can able to enable raw saving. With A better camera I get a corrupted raw file.

**Comment by aggere1** [comment]
@defcomk, given zmax pro has a locked bootloader and I can't hack build.prop, do I have any other options?
